OpenRA.Mods.Common/Widgets/Logic/Ingame/GameTimerLogic.cs

```diff
@@ -51,7 +52,10 @@ public GameTimerLogic(Widget widget, OrderManager orderManager, World world)
 					if (status == null && shouldShowStatus())
 						return statusText();
 
-					return WidgetUtils.FormatTime(world.WorldTick, timestep);
+					var timeLimit = tlm != null ? tlm.TimeLimit : 0;
+					return timeLimit > 0 ?
+						WidgetUtils.FormatTime(timeLimit - world.WorldTick, world.Timestep) :
```

Comment:
This breaks the timer when replaying at speeds other than 100%.  Change `world.Timestep` to `timestep` to fix this.